app/vmselect/promql/eval.go

```diff
@@ -815,6 +816,14 @@ func evalRollupFunc(qt *querytracer.Tracer, ec *EvalConfig, funcName string, rf
 		}
 	}
 	atTimestamp := int64(tssAt[0].Values[0] * 1000)
+	// Find the first non-NaN value in tssAt.
+	if math.IsNaN(tssAt[0].Values[0]) {
+		for _, v := range tssAt[0].Values {
+			if !math.IsNaN(v) {
+				atTimestamp = int64(v * 1000)
+			}
+		}
+	}
```

Comment:
```suggestion
	// Find the first non-NaN value in tssAt.
        atValue := math.NaN
	for _, v := range tssAt[0].Values {
		if !math.IsNaN(v) {
			atValue = v
		}
	}
	if math.IsNaN(atValue) {
	  // return err
	}
       atTimestamp := int64(atValue * 1000)
```

Reply:
maybe wrap it in a function if it doesn't look elegant.
I'd also like to see tests where this can be reproduced.

Reply:
It seems like currently the only option would be to create an integration test for this, would that work?

Reply:
yes, it would be great